mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -69,21 +73,26 @@ class ChiSqSelectorModel @Since("1.3.0") (
    * Preserves the order of filtered features the same as their indices are stored.
    * Might be moved to Vector as .slice
    * @param features vector
-   * @param filterIndices indices of features to filter, must be ordered asc
+   * @param filterIndices indices of features to filter
    */
   private def compress(features: Vector, filterIndices: Array[Int]): Vector = {
+    val orderedIndices = if (isSorted(filterIndices)) {
```

Comment:
I don't think there's any point in checking whether it's sorted. `.sorted` would ultimately do about the same thing and do no work if sorted already, so it doesn't add much. Plus you can remove the isSorted method too.